aiocache/backends/valkey.py

```diff
@@ -126,16 +132,43 @@ async def _expire(self, key, ttl, _conn=None):
         return await self.client.expire(key, ttl)
 
     async def _delete(self, key, _conn=None):
+        if isinstance(key, str):
+            key = [key]
         return await self.client.delete(key)
 
     async def _clear(self, namespace=None, _conn=None):
         if namespace:
-            keys = await self.client.keys("{}:*".format(namespace))
+            cursor, keys = await self.client.scan(b"0", "{}:*".format(namespace))
             if keys:
-                await self.client.delete(*keys)
+                return bool(await self.client.delete(keys))
         else:
-            await self.client.flushdb()
-        return True
```

Comment:
This return has gone missing. if namespace and not keys would miss any return statements.